docs/src/model_simulation/examples/periodic_events_simulation.md

```diff
@@ -0,0 +1,95 @@
+# [Modelling a periodic event during ODE and jump simulations](@id periodic_event_simulation_example)
+This tutorial will describe how to simulate systems with periodic events in ODE (straightforward) and jump (slightly less, but still fairly, straightforward) simulations (SDEs use identical syntax to ODEs). We will consider a model with a [circadian rhythm](https://en.wikipedia.org/wiki/Circadian_rhythm), where a parameter represents the level of light. While outdoor light varies smoothly, in experimental settings a lamp is often simply turned on/off every 12 hours. Here we will model this toggling of the light using a periodic event that is triggered every 12 hours.
+
+## [Modelling a circadian periodic event in an ODE simulation](@id periodic_event_simulation_example_ode)
+We will consider a simple circadian model, consisting of a single protein ($X$), which is phosphorylated ($X \to Xᴾ$) in the presence of light ($l$). Here, the light parameter can either be $0$ (night) or $1$ (day). We can model this using a simple periodic event which switches the value of $l$ every 12 hours (here, `%` is the [remainder operator](https://docs.julialang.org/en/v1/manual/mathematical-operations/#Arithmetic-Operators)).
+```@example periodic_event_example
+using Catalyst
+circadian_model = @reaction_network begin
+    @discrete_events begin
+        12 => [l ~ (l + 1)%2]
```

Comment:
MTK's events really need some attention. I really don't like this having to hardcode a parameter value (the event time) within the callback. We need a way to specify periodic and preset using symbolic variables to encode times (along with a way to avoid users having to also specify `tstop` values manually for discrete events).

Reply:
I guess there is a problem what to do if the parameter(s) that set the time is updated mid-simulation, but once it is decided how to deal with that it should be straightforward.